internal/cmd/kafka/command_cluster_update.go

```diff
@@ -66,31 +74,97 @@ func (c *clusterCommand) update(cmd *cobra.Command, args []string, prompt form.P
 			return err
 		}
 		if name == "" {
-			return errors.New(errors.NonEmptyNameErrorMsg)
+			return errors.New("`--name` must not be empty")
 		}
+
 		update.Spec.SetDisplayName(name)
+
+		// The backend blocks simultaneous modification of `--name` and `--type`. If both are passed, update the name separately.
+		if cmd.Flags().Changed("type") {
+			if _, err := c.V2Client.UpdateKafkaCluster(clusterId, update); err != nil {
+				return err
+			}
+			update.Spec.DisplayName = nil
+		}
 	}
 
-	if cmd.Flags().Changed("cku") {
-		cku, err := cmd.Flags().GetUint32("cku")
+	if cmd.Flags().Changed("availability") {
+		availability, err := cmd.Flags().GetString("availability")
 		if err != nil {
 			return err
 		}
-		updatedCku, err := c.validateResize(int32(cku), &currentCluster, prompt)
-		if err != nil {
-			return err
+		if availability == "" {
+			return errors.New("`--availability` must not be empty")
 		}
-		update.Spec.Config = &cmkv2.CmkV2ClusterSpecUpdateConfigOneOf{CmkV2Dedicated: &cmkv2.CmkV2Dedicated{Kind: "Dedicated", Cku: updatedCku}}
+		update.Spec.SetAvailability(strings.ToUpper(strings.ReplaceAll(availability, "-", "_")))
 	}
 
-	updatedCluster, err := c.V2Client.UpdateKafkaCluster(clusterID, update)
+	// Both `--type` and `--cku` require changing the config
+	if cmd.Flags().Changed("type") || cmd.Flags().Changed("cku") {
```

Comment:
Can we consolidate the logic to make it more readable? I was thinking something like:
```
if (nameChange and clusterTypeChange) {
  setName = new_name
  updateName
  setName = nil
  setClusterType
}

if (ckuChange) {
  updateCKU 
}
```